src/os/bluestore/bluestore_tool.cc

```diff
@@ -780,45 +778,52 @@ int main(int argc, char **argv)
       cerr << "can't allocate new WAL device, already exists"
 	    << std::endl;
       exit(EXIT_FAILURE);
-    } else if(!dev_target.empty() &&
-	      (rlpath = realpath(dev_target.c_str(), target_path)) == nullptr) {
-      cerr << "failed to retrieve absolute path for " << dev_target
-           << ": " << cpp_strerror(errno)
-           << std::endl;
-      exit(EXIT_FAILURE);
     }
 
+    auto [target_path, has_size_spec] =
+      [&dev_target]() -> std::pair<string, bool> {
+      if (dev_target.empty()) {
+	return {"", false};
+      }
+      std::error_code ec;
+      fs::path target_path = fs::weakly_canonical(dev_target, ec);
```

Comment:
```
-- The CXX compiler identification is GNU 7.5.0
-- The C compiler identification is GNU 7.5.0
...
/build/ceph-17.0.0-4436-g0a8ba2e2/src/os/bluestore/bluestore_tool.cc:789:34: error: 'weakly_canonical' is not a member of 'fs'
       fs::path target_path = fs::weakly_canonical(dev_target, ec);
                                  ^~~~~~~~~~~~~~~~
```

turns out libstdc++ shipped along with GCC 7.5.0 does not offer `std::filesystem::weakly_canonical()` yet. and we cannot drop bionic yet. see https://github.com/ceph/ceph-build/pull/1794#issuecomment-815435506